src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -52,7 +52,7 @@ public static string FindSolutionRoot()
             string path = null;
             try
             {
-                ThreadHelper.Generic.Invoke(() => { path = Internal.FindSolutionRootImpl(); });
+                ThreadHelper.JoinableTaskFactory.Run(async () => { path = Internal.FindSolutionRootImpl(); });
```

Comment:
This is *not* equivalent. 
`ThreadHelper.Generic.Invoke` invokes the delegate on the UI thread.
`JoinableTaskFactory.Run` invokes the delegate on the caller's thread.
If you need `FindSolutionRootImpl` to run on the main thread, you should change it to this:

```cs
ThreadHelper.JoinableTaskFactory.Run(async () => { 
   await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
   path = Internal.FindSolutionRootImpl();
});
```